code/game/machinery/computer/prisoner.dm

```diff
@@ -24,118 +21,149 @@
-			dat += text("Space Law recommends sentences of 150 points per minute they would normally serve in the brig.<BR>")
-		else
-			dat += text("<A href='?src=[UID()];id=0'>Insert Prisoner ID</A><br>")
-		var/turf/Tr = null
-		dat += "<HR>Chemical Implants<BR>"
-		for(var/obj/item/implant/chem/C in GLOB.tracked_implants)
-			Tr = get_turf(C)
-			if((Tr) && (Tr.z != src.z))	continue//Out of range
-			if(!C.implanted) continue
-			// AUTOFIXED BY fix_string_idiocy.py
-			// C:\Users\Rob\Documents\Projects\vgstation13\code\game\machinery\computer\prisoner.dm:41: dat += "[C.imp_in.name] | Remaining Units: [C.reagents.total_volume] | Inject: "
-			dat += {"[C.imp_in.name] | Remaining Units: [C.reagents.total_volume] | Inject:
-				<A href='?src=[UID()];inject1=\ref[C]'>(<font color=red>(1)</font>)</A>
-				<A href='?src=[UID()];inject5=\ref[C]'>(<font color=red>(5)</font>)</A>
-				<A href='?src=[UID()];inject10=\ref[C]'>(<font color=red>(10)</font>)</A><BR>
-				********************************<BR>"}
-			// END AUTOFIX
-		dat += "<HR>Tracking Implants<BR>"
-		for(var/obj/item/implant/tracking/T in GLOB.tracked_implants)
-			Tr = get_turf(T)
-			if((Tr) && (Tr.z != src.z))	continue//Out of range
-			if(!T.implanted) continue
-			var/mob/living/carbon/M = T.imp_in
-			var/loc_display = "Unknown"
-			var/health_display = "OK"
-			var/total_loss = (M.maxHealth - M.health)
-			if(M.stat == DEAD)
-				health_display = "DEAD"
-			else if(total_loss)
-				health_display = "HURT ([total_loss])"
-			if(is_station_level(Tr.z) && !istype(Tr.loc, /turf/space))
-				loc_display = "[get_area(Tr)]"
-			dat += "ID: [T.id] <BR>Subject: [M] <BR>Location: [loc_display] <BR>Health: [health_display] <BR>"
-			dat += "<A href='?src=[UID()];warn=\ref[T]'>(<font color=red><i>Message Holder</i></font>)</A> |<BR>"
-			dat += "********************************<BR>"
-		dat += "<HR><A href='?src=[UID()];lock=1'>Lock Console</A>"
-
-	user << browse(dat, "window=computer;size=400x500")
-	onclose(user, "computer")
-	return
-
-/obj/machinery/computer/prisoner/process()
-	if(!..())
-		src.updateDialog()
-	return
-
-/obj/machinery/computer/prisoner/Topic(href, href_list)
+		return TRUE
+
+	ui_interact(user)
+
+/obj/machinery/computer/prisoner/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "PrisonerImplantManager", name, 475, 500, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/prisoner/ui_data(mob/user)
+	var/list/data = list()
+	ui_login_data(data, user)
+	data["prisonerInfo"] = list(
+		"name" = inserted_id?.name,
+		"points" = inserted_id?.points,
+		"goal" = inserted_id?.goal,
+	)
+
+	data["chemicalInfo"] = list()
+	for(var/obj/item/implant/chem/C in GLOB.tracked_implants)
+		var/turf/implant_location = get_turf(C)
```

Comment:
These checks should probably be made into a proc since you're using them twice. Once for the chem implants. And once for the tracking implants

Reply:
done